File: CS-Examples/09_Charts/AdjustBarSpace.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Charts;

namespace AdjustBarSpace
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartSample1.xlsx");

            //Get the first worksheet from workbook and then get the first chart from the worksheet
            Worksheet ws = workbook.Worksheets[0];
            Chart chart = ws.Charts[0];

            //Adjust the space between bars
            foreach (ChartSerie cs in chart.Series)
            {
                cs.Format.Options.GapWidth = 200;
                cs.Format.Options.Overlap = 0;
            }

            //Save the document
            string output = "AjustBarSpace.xlsx";
			workbook.SaveToFile(output, ExcelVersion.Version2013);

            //Launch the file
			ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/09_Charts/AdjustBarSpace.vb
Imports Spire.Xls
Imports Spire.Xls.Charts

Namespace AdjustBarSpace

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "ChartSample1.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartSample1.xlsx")

            ' Get the first worksheet from the workbook
            Dim ws As Worksheet = workbook.Worksheets(0)

            ' Get the first chart from the worksheet
            Dim chart As Chart = ws.Charts(0)

            ' Iterate through each series in the chart
            For Each cs As ChartSerie In chart.Series
                ' Set the gap width of the series to 200
                cs.Format.Options.GapWidth = 200
                ' Set the overlap of the series to 0
                cs.Format.Options.Overlap = 0
            Next cs

            ' Specify the output file name as "AdjustBarSpace.xlsx"
            Dim output As String = "AdjustBarSpace.xlsx"

            ' Save the modified workbook to the specified file path, using the Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
